From 03e59a3fec5895281917775e75a843f1ee058cbf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?= <avarab@gmail.com>
Date: Sat, 26 Jun 2010 12:11:46 +0000
Subject: [PATCH] t9700: Use Test::More->builder, not $Test::Builder::Test
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

$Test::Builder::Test was only made into an `our' variable in 0.94
released in September 2009, older distros are more likely to have 0.92
or earlier. Use the singleton Test::More->builder constructor instead,
the test now works on 0.92 too.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---

This patch fixes the TAP output in pu for older Test::More
libraries. It could be applied as a fixup to "test-lib: Make the
test_external_* functions TAP-aware", but it's probably useful to note
this in the version history.

 t/t9700/test.pl |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/t/t9700/test.pl b/t/t9700/test.pl
index e5d4b03..3e6a27c 100755
--- a/t/t9700/test.pl
+++ b/t/t9700/test.pl
@@ -10,8 +10,8 @@ use Test::More qw(no_plan);
 BEGIN {
 	# t9700-perl-git.sh kicks off our testing, so we have to go from
 	# there.
-	$Test::Builder::Test->{Curr_Test} = 1;
-	$Test::Builder::Test->{No_Ending} = 1;
+	Test::More->builder->{Curr_Test} = 1;
+	Test::More->builder->{No_Ending} = 1;
 }
 
 use Cwd;
@@ -113,6 +113,6 @@ like($last_commit, qr/^[0-9a-fA-F]{40}$/, 'rev-parse returned hash');
 my $dir_commit = $r2->command_oneline('log', '-n1', '--pretty=format:%H', '.');
 isnt($last_commit, $dir_commit, 'log . does not show last commit');
 
-printf "1..%d\n", $Test::Builder::Test->{Curr_Test};
+printf "1..%d\n", Test::More->builder->{Curr_Test};
 
-exit($Test::Builder::Test->{Is_Passing} ? 0 : 1);
+exit(Test::More->builder->{Is_Passing} ? 0 : 1);
-- 
1.7.1.251.g92a7

